dotCMS/src/main/java/com/dotcms/browser/BrowserAPIImpl.java

```diff
@@ -0,0 +1,524 @@
+
+            maxResults = returnList.size() - offset;
+        }
+
+        final Map<String, Object> returnMap = new HashMap<>();
+        returnMap.put("total", returnList.size());
+        returnMap.put("list",  returnList.subList(offset, offset + maxResults));
+        return returnMap;
+    }
+
+    private List<Map<String, Object>> filterReturnList(final BrowserQuery browserQuery, final List<Map<String, Object>> returnList) {
+
+        final List<Map<String, Object>> filteredList = new ArrayList<Map<String, Object>>();
+        for (final Map<String, Object> asset : returnList) {
+
+            String name = (String) asset.get("name");
+            name = name == null ? StringPool.BLANK : name;
+
+            String description = (String) asset.get("description");
+            description = description == null ? StringPool.BLANK : description;
+
+            String mimeType = (String) asset.get("mimeType");
+            mimeType = mimeType == null ? StringPool.BLANK : mimeType;
+
+            if (browserQuery.mimeTypes != null && browserQuery.mimeTypes.size() > 0) {
+
+                boolean match = false;
+                for (final String mType : browserQuery.mimeTypes) {
+                    if (mimeType.contains(mType)) {
+                        match = true;
+                    }
+                }
+
+                if (!match) {
+                    continue;
+                }
+            }
+
+            if (browserQuery.extensions != null && browserQuery.extensions.size() > 0) {
+
+                boolean match = false;
+                for (final String extension : browserQuery.extensions) {
+                    if (((String) asset.get("extension")).contains(extension)) {
+                        match = true;
+                    }
+                }
+
+                if (!match) {
+                    continue;
+                }
+            }
+
+            filteredList.add(asset);
+        }
+
+        return filteredList;
+    }
+
+    private String createQuery(final BrowserQuery browserQuery, final Folder parent, final Host host) {
+
+        final StringBuilder luceneQuery = new StringBuilder();
+        final List<String> baseTypes = new ArrayList<>();
+
+        if(browserQuery.showDotAssets) {
+
+            baseTypes.add(String.valueOf(BaseContentType.DOTASSET.getType()));
+        }
+
+        if(browserQuery.showFiles) {
+
+            baseTypes.add(String.valueOf(BaseContentType.FILEASSET.getType()));
+        }
+
+        if(browserQuery.showPages) {
+
+            baseTypes.add(String.valueOf(BaseContentType.HTMLPAGE.getType()));
+        }
+
+        if(baseTypes.size()==0) {
+
+            baseTypes.add(String.valueOf(BaseContentType.ANY.getType()));
+        }
+
+        luceneQuery.append("+basetype:(" + String.join(" OR ", baseTypes) + ") ");
+        luceneQuery.append(browserQuery.languageId > 0? " +languageid:" + browserQuery.languageId : StringPool.BLANK);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [StringBuffer (or StringBuilder).append is called consecutively without reusing the target variable.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)